src/snowflake/connector/connection.py

```diff
@@ -1233,6 +1288,8 @@ def _authenticate(self, auth_instance: AuthByPlugin):
         )
 
         auth = Auth(self.rest)
+        # record start time for computing timeout
+        auth_instance._retry_ctx.set_start_time()
```

Comment:
how do we record the start time in the previous implementation?
also if there is a `set_start_time`, will there be a `unset` method to be called?

Reply:
[Start time was previously set like this.](https://github.com/snowflakedb/snowflake-connector-python/blob/4777d83f8cac79194f890b34e50162e35478ee00/src/snowflake/connector/network.py#L887)

I added a reset on `start_time_millis` in the `reset` method of the context.